src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,95 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Represents a parser that will be used to parse each line of input
+ */
+public class Parser {
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Constructs a Parser
+     * @param taskList taskList to which the parsed Tasks would be saved
+     * @param ui ui for input and output
+     */
+    public Parser(TaskList taskList, Ui ui) {
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the inputted string into a Task
+     * @param  str string to be parsed
+     */
+    public void parseLine(String str) throws DukeException, NumberFormatException{
+        if (str.equals("list")) {
+            ui.printList();
+        } else if (str.startsWith("delete ")) {
```

Comment:
Never knew the String method .startsWith() existed! To check input, I created a Scanner object and would read the first word of the input which made my code very messy. Your method makes code very readable, learnt something new!